src/components/QuranReader/index.tsx

```diff
@@ -38,18 +40,19 @@ const verseFetcher = async (input: RequestInfo, init?: RequestInit) => {
 
 const QuranReader = ({ initialData, chapter }: QuranReaderProps) => {
   const quranReaderStyles = useSelector(selectQuranReaderStyles);
+  const currentTranslations = useSelector(selectCurrentTranslations) as string[];
   const { data, size, setSize, isValidating } = useSWRInfinite(
-    (index) => {
-      // TODO: select the translation using the user preference
-      return makeVersesUrl(chapter.id, {
+    (index) =>
+      makeVersesUrl(chapter.id, {
         page: index + 1,
         wordFields: `verse_key, verse_id, page_number, location, ${quranReaderStyles.quranFont}`,
-      });
-    },
+        translations: joinStringArray(currentTranslations),
+      }),
     verseFetcher,
     {
       initialData: initialData.verses,
-      revalidateOnFocus: true,
+      revalidateOnFocus: false, // disable auto revalidation when window gets focused
+      revalidateOnMount: true, // enable automatic revalidation when component is mounted. This is needed when the translations inside initialData don't match with the user preferences and would result in inconsistency either when we first load the QuranReader with pre-saved translations from the persistent store or when we change the translations' preferences after initial load.
```

Comment:
does revalidate only rerenders if the data is different? 

If not, we'll need our own custom logic to prevent do the revalidation conditionally. Instead of setting it to always be `true`